handler/src/main/java/com/networknt/handler/HandlerUtils.java

```diff
@@ -68,4 +72,15 @@ public static String toInternalKey(String method, String path) {
         return String.format(INTERNAL_KEY_FORMAT, method, HandlerUtils.normalisePath(path));
     }
 
+    public static void populateAuditAttachmentField(final HttpServerExchange exchange, String fieldName, String fieldValue) {
+        Map<String, Object> auditInfo = exchange.getAttachment(AttachmentConstants.AUDIT_INFO);
+        if(auditInfo == null) {
+            auditInfo = new HashMap<>();
+            auditInfo.put(fieldName, fieldValue);
+        } else if (!auditInfo.containsKey(Constants.ENDPOINT_STRING)) {
```

Comment:
Should Constants.ENDPOINT_STRING be fieldName? Although we know this method is used to populate the endpoint, we must allow other users to use the same method to populate other fields.